src/Tezos/Forging.h

```diff
@@ -8,14 +8,64 @@
 
 #include "../PublicKey.h"
 #include "../proto/Tezos.pb.h"
+#include "uint256.h"
 
 #include <string>
+#include <vector>
+#include <variant>
 
 using namespace TW;
 using namespace TW::Tezos::Proto;
 
+enum PrimType : std::uint8_t {
+    Pair = 7,
+};
+
+struct MichelsonValue;
+
+struct PrimValue {
+    std::string prim;
+    std::vector<MichelsonValue> args;
+    std::vector<std::string> anots;
+};
+
+struct BytesValue {
+    std::string bytes;
+};
+
+struct StringValue {
+    std::string string;
+};
+
+struct IntValue {
+    std::string _int;
+};
+
+struct MichelsonValue {
+    using MichelsonArray = std::vector<std::variant<PrimValue, BytesValue, StringValue, IntValue>>;
+    using MichelsonVariant = std::variant<
+        PrimValue,
+        BytesValue,
+        StringValue,
+        IntValue,
+        MichelsonArray>;
+    MichelsonVariant value;
+};
+
+
+MichelsonValue::MichelsonVariant FA12ParameterToMichelson(const FA12TransactionOperationData& data);
+MichelsonValue::MichelsonVariant FA2ParameterToMichelson(const FA2TransactionOperationData& data);
+
 Data forgeBool(bool input);
 Data forgeOperation(const Operation& operation);
+Data forgeAddress(const std::string& address);
+Data forgeArray(const Data& data, int len = 4);
```

Comment:
Same question for len -- is it number of bytes to use to encode array size?

Reply:
It can be used for other instructions, for the moment as you can see we only support `Pair` in `PrimType`, but actually many other instructions are possible, we can remove it, but it's a bit more flexible with it.

You can see the full type list here: https://github.com/baking-bad/netezos/blob/976c0c02d4ac483c70b7f600f31bbeefdc451a3f/Netezos/Encoding/Micheline/Abstract/PrimType.cs#L7

And the spec: 

- https://tezos.gitlab.io/shell/p2p_api.html
- https://tezos.gitlab.io/active/michelson.html